drivers/display/display_sdl.c

```diff
@@ -54,8 +63,12 @@ static int sdl_display_init(const struct device *dev)
 #endif /* SDL_DISPLAY_DEFAULT_PIXEL_FORMAT */
 		;
 
+	if (sdl_display_zoom_pct == UINT32_MAX) {
+		sdl_display_zoom_pct = CONFIG_SDL_DISPLAY_ZOOM_PCT;
+	}
```

Comment:
Why is this check needed?

Reply:
Actually, if I understood well, if you declare an option, it will take a default value, as set in https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/native_simulator/native/src/nsi_cmdline_common.c#L185

Unless you override it with the CLI argument, it will stay that way.

I also kept a [default value](https://github.com/zephyrproject-rtos/zephyr/pull/65747/files#diff-bbd24d917525226ebac65891ea3a1d7b7db14d284837f1ebc07b5e50e9d38b91R28) is case `display_sdl` is used in a non-POSIX context. Not sure this case exists.

But maybe I'm wrong in my understanding.

Reply:
@arnaudmz your understanding is correct :)

> I also kept a default value is case display_sdl is used in a non-POSIX context. Not sure this case exists.

It doesn't, so that assignment in line 28 is irrelevant as it is.